store/event_history.go

```diff
@@ -83,12 +83,12 @@ func (eh *EventHistory) scan(key string, recursive bool, index uint64) (*Event,
 
 			if recursive {
 				// add tailing slash
-				key = path.Clean(key)
-				if key[len(key)-1] != '/' {
-					key = key + "/"
+				nkey := path.Clean(key)
```

Comment:
looking at this again, I think this only needs to be computed once, before the for loop?